### ISSUE TITLE
Implement cosmetic (css/js/extcss) rules logic

### ISSUE THREAD

**Issue by hmage** [AdguardTeam/CoreLibs#50]
Subj

**Comment by hmage** [comment]
JNI done, now java part remains.

**Comment by ameshkov** [comment]
`testGenericHideRule` failed

**Comment by ameshkov** [comment]
`testUblockCssInjectionSyntaxSupport` failed

**Comment by ameshkov** [comment]
`testCssInjectionRules` failed

**Comment by hmage** [comment]
All failing java tests are fixed in commit 06b0d84c8033f89586af6bbda185ee1f84d8714c.

**Comment by ameshkov** [comment]
`testExtendedCssGenericRules` failed.

The thing is that ExtCss rules shouldn't be grouped like we do it for regular CSS rules.

Regular: 
`selector1, selector2 { style }`

ExtCss: `selector1 { style } selector2 { style }`

The fix should be more or less trivial.

**Comment by ameshkov** [comment]
Fixed

**Comment by ameshkov** [comment]
Tests for `NativeJsFilter` fail, there's garbage in the end of the `buildScript` result

**Comment by ameshkov** [comment]
Added line termination:
```
    if (result.used) {
        if (!add_to_str(&result, "", 1)) {
            return NULL;
        }
    }
```

**Comment by hmage** [comment]
Was merged in commit 5c4bebeae866ef9bd60370209a387f1ea389a878.
